quit/git.py

```diff
@@ -0,0 +1,638 @@
+import os
+import pygit2
+import re
+
+from datetime import datetime
+from exceptions import RepositoryNotFound, RevisionNotFound, NodeNotFound
+from rdflib import Graph, Literal, URIRef, ConjunctiveGraph, Dataset, BNode
+from quit.namespace import FOAF, RDFS, PROV, QUIT, is_a, XSD
+from quit.utils import graphdiff, clean_path
+
+# roles
+role_author = QUIT['author']
+role_committer = QUIT['committer']
+
+def _git_timestamp(ts, offset):
+    import quit.utils as tzinfo
+    if offset == 0:
+        tz = utc
+    else:
+        hours, rem = divmod(abs(offset), 60)
+        tzname = 'UTC%+03d:%02d' % ((hours, -hours)[offset < 0], rem)
+        tz = tzinfo.TZ(offset, tzname)
+    return datetime.fromtimestamp(ts, tz)
+
+class Base(object):
+    def __init__(self):
+        pass
+
+    def __prov__(self):
+        pass
+
+
+class Repository(Base):
+    def __init__(self, path, **params):
+        origin = params.get('origin', None)
+
+        try:
+            self._repository = pygit2.Repository(path)
+        except KeyError:
+            if not params.get('create', False):
+                raise RepositoryNotFound('Repository "%s" does not exist' % path)
+                        
+            if origin:
+                self.callback = self._callback(origin)
+                pygit2.clone_repository(url=origin, path=path, bare=False)
+            else:
+                pygit2.init_repository(path)
+
+        name = os.path.basename(path).lower()
+
+        self.name = name
+        self.path = path
+        self.params = params
+
+    def _callback(self, origin):
```

Comment:
I think we have a better solution for this already here: https://github.com/AKSW/QuitStore/blob/ff4598a4ca6dbe83619924df67d3d4f2816e5f58/quit/core.py#L654L697